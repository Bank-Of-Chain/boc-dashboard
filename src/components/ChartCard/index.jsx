import { Card } from "antd";
import React from "react";
import classNames from "classnames";
import styles from "./index.less";

const renderTotal = (total, unit) => {
  if (!total && total !== 0) {
    return null;
  }

  let totalDom;

  switch (typeof total) {
    case "undefined":
      totalDom = null;
      break;

    case "function":
      totalDom = (
        <div className={styles.total}>
          <span>{total()}</span>
          {unit && <span className={styles.unit}>{unit}</span>}
        </div>
      );
      break;

    default:
      totalDom = (
        <div className={styles.total}>
          <span>{total}</span>
          {unit && <span className={styles.unit}>{unit}</span>}
        </div>
      );
  }

  return totalDom;
};

class ChartCard extends React.Component {
  renderContent = () => {
    const {
      contentHeight,
      title,
      avatar,
      action,
      total,
      footer,
      children,
      loading,
      unit,
    } = this.props;

    if (loading) {
      return false;
    }

    return (
      <div className={styles.chartCard}>
        <div className={classNames(styles.chartTop)}>
          <div className={styles.avatar}>{avatar}</div>
          <div className={styles.metaWrap}>
            <div className={styles.meta}>
              <span className={styles.title}>{title}</span>
              <span className={styles.action}>{action}</span>
            </div>
            {renderTotal(total, unit)}
          </div>
        </div>
        {children && (
          <div
            className={styles.content}
            style={{
              height: contentHeight || "auto",
            }}
          >
            <div className={contentHeight && styles.contentFixed}>
              {children}
            </div>
          </div>
        )}
        {footer && (
          <div
            className={classNames(styles.footer, {
              [styles.footerMargin]: !children,
            })}
          >
            {footer}
          </div>
        )}
      </div>
    );
  };

  render() {
    const {
      loading = false,
      contentHeight,
      title,
      avatar,
      action,
      total,
      footer,
      children,
      ...rest
    } = this.props;
    return (
      <Card
        loading={loading}
        bodyStyle={{
          padding: "2.375rem 2.125rem 2.375rem 2.75rem",
        }}
        {...rest}
      >
        {this.renderContent()}
      </Card>
    );
  }
}

export default ChartCard;
