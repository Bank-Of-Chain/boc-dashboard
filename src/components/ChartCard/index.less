@import "~antd/es/style/themes/default.less";

.chartCard {
  position: relative;
  .title {
    color: @text-color;
    font-size: 1.25rem;
    line-height: 1.875rem;
  }
  .chartTop {
    position: relative;
    width: 100%;
    overflow: hidden;
  }
  .chartTopMargin {
    margin-bottom: 12px;
  }
  .chartTopHasMargin {
    margin-bottom: 20px;
  }
  .metaWrap {
    float: left;
  }
  .avatar {
    position: relative;
    top: 4px;
    float: left;
    margin-right: 20px;
    img {
      border-radius: 100%;
    }
  }
  .meta {
    color: @text-color-secondary;
    font-size: @font-size-base;
  }
  .action {
    position: absolute;
    top: 4px;
    right: 0;
    line-height: 1;
    cursor: pointer;
  }
  .total {
    margin-top: 0.625rem;
    margin-bottom: 0;
    overflow: hidden;
    color: @text-color;
    font-weight: bold;
    font-size: 2.5rem;
    line-height: 3.25rem;
    white-space: nowrap;
    text-overflow: ellipsis;
    word-break: break-all;
    .unit {
      margin-left: 0.5rem;
      font-size: 2.375rem;
      background-image: linear-gradient(
        223.3deg,
        #a68efd 20.71%,
        #f4acf3 103.56%
      );
      background-clip: text;
      -webkit-text-fill-color: transparent;
    }
  }
  .content {
    position: relative;
    width: 100%;
  }
  .contentFixed {
    position: absolute;
    bottom: 0;
    left: 0;
    width: 100%;
  }
  .footer {
    margin-top: 0.5rem;
    padding-top: 0.5rem;
    border-top: 1px solid @border-color-split;
    & > * {
      position: relative;
    }
  }
  .footerMargin {
  }
}

.card {
  :global {
    .ant-card-body {
      padding: 2.375rem 2.125rem 2.375rem 2.75rem;
    }
  }
}

@media screen and (max-width: 554px) {
  .card {
    :global {
      .ant-card-body {
        padding: 1rem;
      }
    }
  }

  .chartCard {
    .action {
      right: 2rem;
    }
    .footer {
      margin-top: 0;
      padding-top: 0.5rem;
    }
  }
}
