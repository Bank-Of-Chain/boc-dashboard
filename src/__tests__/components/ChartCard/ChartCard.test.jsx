import React from 'react'
import { render } from '@testing-library/react'
import ChartCard from '@/components/ChartCard/index'

test('ChartCard Component Render', () => {
  const { asFragment } = render(<ChartCard loading={false} title={'test text'} />)
  expect(asFragment()).toMatchSnapshot()
})
