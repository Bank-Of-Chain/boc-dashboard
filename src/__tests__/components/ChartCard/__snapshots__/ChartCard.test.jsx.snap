// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChartCard Component Render 1`] = `
<DocumentFragment>
  <div
    class="ant-card ant-card-bordered p-4 b-rd-5 css-dev-only-do-not-override-ed5zg0"
  >
    <div
      class="ant-card-body"
    >
      <div>
        <div
          class="w-full overflow-hidden"
        >
          <div
            class="text-4 flex justify-between"
          >
            <span
              class="text-5 text-white"
            >
              test text
            </span>
            <span
              class="cursor-pointer"
            />
          </div>
        </div>
        <div
          class="text-3 h-4"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
